tests/mocha/insertion_marker_manager_test.js

```diff
@@ -0,0 +1,329 @@
+            const manager = createBlocksAndManager(this.workspace, state);
+            const markers = manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 2);
+        });
+
+        test('One value block', function() {
+            const state = {
+              'blocks': {
+                'blocks': [
+                  {
+                    'type': 'row_block',
+                    'id': 'first',
+                  },
+                ],
+              },
+            };
+            const manager = createBlocksAndManager(this.workspace, state);
+            const markers = manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 1);
+        });
+
+        test('Two value blocks', function() {
+            const state = {
+              'blocks': {
+                'blocks': [
+                  {
+                    'type': 'row_block',
+                    'id': 'first',
+                    'inputs': {
+                        'INPUT': {
+                          'block': {
+                            'type': 'row_block',
+                            'id': 'second',
+                          },
+                        },
+                      },
+                  },
+                ],
+              },
+            };
+            const manager = createBlocksAndManager(this.workspace, state);
+            const markers = manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 1);
+        });
+    });
+
+    suite('Would delete block', function() {
+        setup(function() {
+            const state = {
+                'blocks': {
+                  'blocks': [
+                    {
+                      'type': 'stack_block',
+                      'id': 'first',
+                    },
+                  ],
+                },
+              };
+            Blockly.serialization.workspaces.load(state, this.workspace);
+            this.block = this.workspace.getBlockById('first');
+            this.manager = new Blockly.InsertionMarkerManager(this.block);
+        });
+        
+        function stubComponentManager(workspace) {
+            const componentManager = workspace.getComponentManager();
+            const stub = sinon.stub(componentManager, 'hasCapability');
+            return stub;
+        }
+
+        test('Over delete area: accepted', function() {
+            const dxy = new Blockly.utils.Coordinate(0, 0);
+            const stub = stubComponentManager(this.workspace);
+            stub.withArgs('fakeDragTarget',
+                Blockly.ComponentManager.Capability.DELETE_AREA).returns(true);
+            const fakeDragTarget = {
+                wouldDelete: sinon.fake.returns(true),
+                id: 'fakeDragTarget',
+            };
+            this.manager.update(dxy, fakeDragTarget);
+            chai.assert.isTrue(this.manager.wouldDeleteBlock());
+            chai.assert.isTrue(fakeDragTarget.wouldDelete.called);
```

Comment:
Is it a promise we've made to external developers that this method will get called? If yes, sweet! If not, I think this just makes the test more of a change-detector, and it should be removed.

Reply:
Good point. I was using it to confirm to myself that the test was actually running correctly, but it doesn't need to stay